main/solution/backend/config/infra/cloudformation.yml

```diff
@@ -767,6 +740,68 @@ Resources:
               - appstream:UpdateImagePermissions
             Resource:
               - !Sub 'arn:aws:appstream:${AWS::Region}:${AWS::AccountId}:image/ServiceWorkbench*'
+          - Effect: Allow
+            Action:
+              - iam:CreateRole
+            Resource:
+              - !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:role/study-*'
+            Condition:
+              StringEquals:
+                iam:PermissionsBoundary: !Ref PermissionBoundaryPolicyStudyBucket
+          - Effect: Allow
+            Action:
+              - iam:AttachRolePolicy
+            Resource:
+              - !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:role/study-*'
+          - Effect: Allow
+            Action:
+              - iam:CreatePolicy
+            Resource:
+              - !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:policy/study-*'
+
+  PermissionBoundaryPolicyStudyBucket:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
```

Comment:
I didn't add a condition here to only create this Policy if `EnableEgressStore` is true, because then we would have to do additional `if` statements on line 748. I wasn't able to get the CF template to compile successfully with an if Statement for line 748. In terms of value given effort spent, I think it should be ok to not add additional `if` statements for the Egress state, given that we will soon transition internal studies to use roles as well. When that happens this role and policy will be needed for both Egress/NonEgress state.

Reply:
If EnableEgressStore is false,  it just looks odd to have a policy created in main account for a bucket that doesn't even exist. Couple of things to try:

1. We either add an If/Else condition on the resource itself on this permission boundary policy. We can go with either of the following approaches here: 
    * (Preferable for now but not sure if possible) When egressStore is enabled we allow read/write access to egress bucket and when egressStore is disabled our permission boundary will be empty suggesting that the IAM role created using the permission boundary doesn't have any access.
    * (Includes what we want to do in future eventually) When egressStore is enabled we allow read/write access to egress bucket and only read access to study bucket. When egressStore is disabled our permission boundary will only have read/write access to study bucket

2. I know it is hard to get nested If conditions in CFN templates. Have you tried applying the If condition for line 743 instead? I had to do something similar for account-onboard yaml template [here](https://github.com/awslabs/service-workbench-on-aws/blob/feat-secure-workspace-egress/addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml#L265-L275). The main difference is that in my case I had a condition for an inline policy and what you have here is a managed policy. Not sure if you already tried this.

Reply:
I tried doing Option 1.a, but that didn't work since with `egressStore`disabled there was an empty policy and CF didn't like it. 

I'm going to go with approach 2 instead.